stl/inc/iterator

```diff
@@ -135,15 +141,26 @@ public:
 
     constexpr istream_iterator() : _Myistr(nullptr), _Myval() {}

```

Comment:
> The cleaner solution would certainly be to utilize member initialization and simply default the default constructor. However I assume that this is not a backward compatible change.

I agree that it's not backward compatible. Most types won't notice the difference, except for poorly-designed types whose stream extractor doesn't fully initialize everything their copy constructor copies. For such types that use the `istream_type&` constructor, the new behavior provides deterministic behavior.

I'd rather treat this like a DR: give `_Myval` a default member initializer (`_Ty _Myval{};`) and simplify the constructors in all language modes.

Please leave this comment open for visibility to other reviewers who may not agree.

Reply:
I also believe that this should be unconditional. We have precedent for making such improvements to existing functionality unconditional; the `enable_shared_from_this` cleanup was similar.

Thanks for being careful to initially submit this in the non-intrusive, conditional form, though!

Reply:
Also, remember to update `yvals_core.h` accordingly (we have a separate section for unconditional features).